tests/foreman/cli/test_host.py

```diff
@@ -2607,6 +2610,212 @@ def test_positive_register_host_ak_with_host_collection(self):
+        })
+        # setup rh satellite tools repository content
+        setup_org_for_a_rh_repo({
+            u'product': PRDS['rhel'],
+            u'repository-set': REPOSET['rhst7'],
+            u'repository': REPOS['rhst7']['name'],
+            u'organization-id': cls.org['id'],
+            u'content-view-id': cls.content_view['id'],
+            u'lifecycle-environment-id': cls.env['id'],
+            u'activationkey-id': cls.activation_key['id'],
+        })
+        # Create custom repository content
+        setup_org_for_a_custom_repo({
+            u'url': FAKE_6_YUM_REPO,
+            u'organization-id': cls.org['id'],
+            u'content-view-id': cls.content_view['id'],
+            u'lifecycle-environment-id': cls.env['id'],
+            u'activationkey-id': cls.activation_key['id'],
+        })
+
+    def setUp(self):
+        """Create VM, install katello-ca package, subscribe vm host, enable
+        satellite-tools repository, and install katello-host-tools package
+        """
+        super(KatelloHostToolsTestCase, self).setUp()
+        # Create VM and register content host
+        self.client = VirtualMachine()
+        self.client.create()
+        self.addCleanup(vm_cleanup, self.client)
+        self.client.install_katello_ca()
+        # Register content host and install katello-host-tools
+        self.client.register_contenthost(
+            self.org['label'],
+            self.activation_key['name'],
+        )
+        self.assertTrue(self.client.subscribed)
+        self.host_info = Host.info({'name': self.client.hostname})
+        self.client.enable_repo(REPOS['rhst7']['id'])
+        self.client.install_katello_host_tools()
+
+    @run_only_on('sat')
+    @tier3
+    def test_positive_report_package_installed_removed(self):
+        """Ensure installed/removed package is reported to satellite
+
+        :id: fa5dc238-74c3-4c8a-aa6f-e0a91ba543e3
+
+        :steps:
+            1. register a host to activation key with content view that contain
+               packages
+            2. install a package 1 from the available packages
+            3. list the host installed packages with search for package 1 name
+            4. remove the package 1
+            5. list the host installed packages with search for package 1 name
+
+        :expectedresults:
+            1. after step3: package 1 is listed in installed packages
+            2. after step5: installed packages list is empty
+
+        :BZ: 1463809
+
+        :CaseLevel: System
+        """
+        self.client.run('yum install -y {0}'.format(FAKE_0_CUSTOM_PACKAGE))
+        result = self.client.run(
+            'rpm -q {0}'.format(FAKE_0_CUSTOM_PACKAGE)
+        )
+        self.assertEqual(result.return_code, 0)
+        installed_packages = Host.package_list({
+            'host-id': self.host_info['id'],
+            'search': 'name={0}'.format(FAKE_0_CUSTOM_PACKAGE_NAME)
+        })
+        self.assertIn(
```

Comment:
please check that `self.assertEqual(len(installed_packages), 1)` or greater than 1, not sure here, but you said always verify len, before check list content, so now I am asking you to do the same =)